modal/mount.py

```diff
@@ -139,6 +139,12 @@ class _MountDir(_MountEntry):
     def description(self):
         return str(self.local_dir.expanduser().absolute())
 
+    def _walk_and_prune(self, top_dir: Path) -> Generator[str, None, None]:
+        for root, dirs, files in os.walk(top_dir, topdown=True):
+            dirs[:] = [d for d in dirs if not self.ignore(Path(os.path.join(root, d)).relative_to(top_dir))]
```

Comment:
😵‍💫 was not obvious to me why this worked (maybe this is a common idiom I'm not familiar with). Could err on the side of over-description and explain in a comment?